File: Longest Consectuive Sequence/step2-using-set.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        longest = 0

        for num in nums:
            if num-1 not in num_set:
                length = 1
                while num+length in num_set:
                    length+=1
                
                longest = max(length, longest)
        
        return longest

File: Longest Consectuive Sequence/step2-using-set.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int>num_set(nums.begin(), nums.end());
        int longest = 0;

        for (auto &n: num_set) {
            if (!num_set.contains(n-1)) {
                int length = 1;
                while (num_set.contains(n + length)) {
                    length += 1;
                }
                longest = max(length, longest);
            }
        }

        return longest;
    }
};
